### PR TITLE
nrf5x: timer: make TIMER0 a TimerAlarm

### PR DESCRIPTION
**Pull request by bradjc**
Co-authored-by: dcellucc <daniel.w.cellucci@nasa.gov>

### Pull Request Overview

This pull request is part of #1237. It reconfigures how the timers for the nRF5x chips. Best I can tell this shouldn't have any affect on existing code, but I couldn't tell you why these changes are all necessary to support 802.15.4.

### Testing Strategy
Copied from #1237 

### TODO or Help Wanted

n/a

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make formatall`.

### CODE REVIEW

File: chips/nrf5x/src/timer.rs [hunk 1]
```diff
@@ -265,10 +265,10 @@ pub struct TimerAlarm {
 
 // CC0 is used for capture
 // CC1 is used for compare/interrupts
-const ALARM_CAPTURE: usize = 0;
-const ALARM_COMPARE: usize = 1;
-const ALARM_INTERRUPT_BIT: registers::Field<u32, Inte::Register> = Inte::COMPARE1;
-const ALARM_INTERRUPT_BIT_SET: registers::FieldValue<u32, Inte::Register> = Inte::COMPARE1::SET;
+const ALARM_CAPTURE: usize = 1;
+const ALARM_COMPARE: usize = 0;
```
Comment (ppannuto) [inline chips/nrf5x/src/timer.rs]: Something is weird here... does that comment need to update, should these constants not have changed?

Comment (bradjc) [reply chips/nrf5x/src/timer.rs]: I don't know. @dcellucci?

Comment (alevy) [reply chips/nrf5x/src/timer.rs]: I'm nearly 100% sure it doesn't matter which is which and the change should either be reverted or the comment should change (and it doesn't matter which one)

Comment (bradjc) [reply chips/nrf5x/src/timer.rs]: ✅

File: chips/nrf5x/src/timer.rs [hunk 2]
```diff
@@ -267,8 +267,8 @@ pub struct TimerAlarm {
 // CC1 is used for compare/interrupts
 const ALARM_CAPTURE: usize = 0;
 const ALARM_COMPARE: usize = 1;
-const ALARM_INTERRUPT_BIT: registers::Field<u32, Inte::Register> = Inte::COMPARE1;
-const ALARM_INTERRUPT_BIT_SET: registers::FieldValue<u32, Inte::Register> = Inte::COMPARE1::SET;
+const ALARM_INTERRUPT_BIT: registers::Field<u32, Inte::Register> = Inte::COMPARE0;
```
Comment (alevy) [inline chips/nrf5x/src/timer.rs]: Well but this needs to be reverted as well, no?

Comment (bradjc) [reply chips/nrf5x/src/timer.rs]: Sure!

**Pull request merged**